test/IdTableTest.cpp

```diff
@@ -276,12 +276,27 @@ TEST(IdTable, push_back_and_assign) {
                     make(i * NUM_COLS + 3), make(i * NUM_COLS + 4)});
     }
 
+    IdTable t2{NUM_COLS, makeAllocator()};
+    // Test the push_back function for spans
+    for (size_t i = 0; i < NUM_ROWS; i++) {
+      std::vector<ValueId> row;
+      row.push_back(Id::makeFromInt(i * NUM_COLS + 1));
+      row.push_back(Id::makeFromInt(i * NUM_COLS + 2));
+      row.push_back(Id::makeFromInt(i * NUM_COLS + 3));
+      row.push_back(Id::makeFromInt(i * NUM_COLS + 4));
+      t2.push_back(row);
+    }
```

Comment:
You can also add a check that the wrong number of columsn throws an error.
(easy for a `CONTRACT_CHECK` and for an `EXPENSIVE_CHECK` you can query if those are currently enabled (see the end of `Exception.h`.